src/devices/QwiicButton/QwiicButton.ButtonStatus.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Iot.Device.QwiicButton.RegisterMapping;
+
+namespace Iot.Device.QwiicButton
+{
+    public sealed partial class QwiicButton
+    {
+        /// <summary>
+        /// Returns whether the button is pressed down.
+        /// </summary>
+        public bool IsPressedDown()
+        {
+            var status = new StatusRegisterBitField(_registerAccess.ReadRegister<byte>(Register.ButtonStatus));
```

Comment:
I'm looking at all these RegisterAccess APIs and I honestly do not like them. If this was using the regular pattern every other device used this line would be just:
`new StatusRegisterBitField(ReadByte(Register.ButtonStatus));`

You need to ask yourself couple of questions:
- is this solution faster? No
- is this solution require writing less code? No/Maybe similar in optimistic case?
- is this solution clearer and easier to understand? No

given the above IMO you should be consistent with other devices and remove the extra abstraction layer

Reply:
I respectfully disagree - I see again and again the same nitty-gritty read/write byte(s) code if I open a random I2C device binding.
It belongs in a common library, and I don't pretend that my I2cRegisterAccess classes are necessarily good enough to be part of such a library - but it's a start.
As professional developers we have a responsibility for following good development pratices, which means among other to reduce code redundancy, ensure separation of concerns and create comprehensive API's.
In my opinion, when creating a new I2C device binding, developers should only have to focus on the actual "business" of the device, i.e. creating a register map, maybe bit field wrappers and then high-level methods that read/write to the registers defined in the register map. Figuring out how to stackalloc, do some bit shifting and other I2C protocol technicalities is not the real "business" of the device. Plus, it makes it harder for new contributers to create device bindings the right way, which is precisely what happened to me.
Anyway, if you don't allow me to keep my I2cRegisterAccess classes within my own device binding, thereby indirectly saying that you are not interested in trying to improve the quality of the codebase, then I don't think that I want my contribution to be part of this repo and I'll just close the pull request.

Reply:
@krwq, before I spend any more time on updating this pull request, please answer my comment right above, so I can decide whether or not I abandon this PR altogether.